crates/tabby/tests/goldentests.rs

```diff
@@ -84,29 +86,50 @@ async fn golden_test(body: serde_json::Value) -> serde_json::Value {
     actual
 }
 
-async fn assert_golden(body: serde_json::Value) {
-    assert_yaml_snapshot!(golden_test(body).await, {
-        ".id" => "test-id"
-    });
+macro_rules! assert_golden {
+    ($expr:expr) => {
+        insta::assert_yaml_snapshot!(golden_test($expr).await, {
+            ".id" => "test-id"
+        });
+    }
 }
 
+#[cfg(all(target_os = "macos", target_arch = "aarch64"))]
 #[tokio::test]
 async fn run_golden_tests() {
-    wait_for_server().await;
+    wait_for_server(Some("metal")).await;
 
-    assert_golden(json!({
+    assert_golden!(json!({
             "language": "python",
             "segments": {
                 "prefix": "def fib(n):\n    ",
                 "suffix": "\n        return fib(n - 1) + fib(n - 2)"
             }
-    }))
-    .await;
+    }));
 
-    assert_golden(json!({
+    assert_golden!(json!({
             "language": "python",
             "segments": {
                 "prefix": "import datetime\n\ndef parse_expenses(expenses_string):\n    \"\"\"Parse the list of expenses and return the list of triples (date, value, currency).\n    Ignore lines starting with #.\n    Parse the date using datetime.\n    Example expenses_string:\n        2016-01-02 -34.01 USD\n        2016-01-03 2.59 DKK\n        2016-01-03 -2.72 EUR\n    \"\"\"\n    for line in expenses_string.split('\\n'):\n        "
             }
-    })).await;
+    }));
+}
+
+#[tokio::test]
```

Comment:
is there a way to ensure the test only runs sequentially within the file? On my system (mac m1/m2), it tries to start `run_golden_tests` and `run_golden_tests_cpu` in parallel, thus caused the port confliction.

Reply:
I'm not quite sure how to do that - especially since I think this conflict would also apply between the `goldentests_chat` file and `goldentests` file, since they both use the same port. We could change the port to fix that, though. I'll see if I can figure out how to run sequentially within an individual file.

Reply:
Attempted this using the `serial_test` crate, could you please rerun and let me know if this is working for you now?

Reply:
> especially since I think this conflict would also apply between the goldentests_chat file and goldentests file,

test between different files are sequential.

> Attempted this using the serial_test crate, could you please rerun and let me know if this is working for you now?
works for me.